### ISSUE TITLE
Wallaby reports false error, mabe a timing bug?

### ISSUE THREAD

**Issue by martinduo** [wallabyjs/public#2554]
### Issue description or question
I think I've found a bug. Let me describe the sequence. I import a json file of a object that defines my data. This object needs some functions that were removed when the object was saved as a json file, so after importing it (which parses it into an object again), I inject the required functions. Then I immediately pass the object to a function in another file, where some of the object's properties are saved for easy reference. One of the functions is used here. But Wallaby flags this as not a function. This is strange since in other tests this function is used with no problem. I'm thinking this is a timing problem, and Wallaby is parsing the code in the receiving file before the injected function is visible. Here's some code snippets to show what is happening.

```
`/**this code is in file1**/
let planObject = samplePlan // from an imported json filet
planObject = addCrumbsPlusHelpers(planObject) // injects the functions
let rootObject = planObject.forecaster // the top level parent object
createPlanObjectReferences(rootObject) // passes the object to another file2`
```

```
`/** this code is in file2**/
function createPlanObjectReferences(planObject) {
  //Create the plan's object references now
  planObject = planObject
  forecaster = planObject
  summary = planObject.summary
  assets = planObject
  persons = planObject.persons
  annuities = planObject.annuities
  accounts = planObject.accounts
  iRAs = planObject.iRAs
  pensions = planObject.pensions
  parametersCollection = planObject.parameters
  parameters = planObject.parameters.params[0] // params is a child array
  let parameters2 = planObject.parameters.theArray()[0] // theArray method returns params
  let bread1 = parameters.breadcrumbs // these both have the same result
  let bread2 = parameters2.breadcrumbs // these both have the same result
}`
```

This code works, but Wallaby says theArray is not a function. In my VSCode debugging mode I can step through the code and inspect the object where I see all the injected functions in the object. The Wallaby output is this.

```
‌0 failing tests, 25 passing  ​Launch Coverage & Test Explorer​ | ​Search Tests​

planObject.parameters.theArray is not a function 
  at createPlanObjectReferences ​src/moduleObjectRoutines.js:73:2​
  at Suite.<anonymous> ​src/_modulePlanObject.test.js:12:2​
  at Object.<anonymous> ​src/_modulePlanObject.test.js:9:0​

Your test suite must contain at least one test. 
  at ​src/_modulePlanObject.test.js:1:53​
```
Note: the error report about my suite must contain one test is a side effect of the error about theArray is not a function. If I comment out the line in file2 with theArray()[0], all error reports go away.

### Wallaby diagnostics report
```
{
  editorVersion: '1.50.1',
  pluginVersion: '1.0.248',
  editorType: 'VSCode',
  osVersion: 'win32 10.0.19041',
  nodeVersion: 'v12.18.3',
  coreVersion: '1.0.982',
  checksum: 'Y2VkNDMzZjNhZjdhMmVjODIxZGYzYzMzMzgyNWY2MDAsMTYwNDUzNDQwMDAwMCwx',
  config: {
    diagnostics: {
      jest: {
        config: {
          configs: [
            {
              automock: false,
              browser: false,
              cache: true,
              cacheDirectory: 'C:\\Users\\Bruce\\AppData\\Local\\Temp\\jest',
              clearMocks: false,
              coveragePathIgnorePatterns: [ '\\\\node_modules\\\\' ],
              cwd: 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend',
              dependencyExtractor: undefined,
              detectLeaks: undefined,
              detectOpenHandles: undefined,
              displayName: undefined,
              errorOnDeprecated: false,
              extraGlobals: [],
              filter: undefined,
              forceCoverageMatch: [],
              globalSetup: undefined,
              globalTeardown: undefined,
              globals: {},
              haste: { computeSha1: false, providesModuleNodeModules: [], throwOnModuleCollision: false },
              moduleDirectories: [ 'node_modules' ],
              moduleFileExtensions: [ 'js', 'json', 'jsx', 'ts', 'tsx', 'node' ],
              moduleLoader: undefined,
              moduleNameMapper: [],
              modulePathIgnorePatterns: [],
              modulePaths: undefined,
              name: '8cc8ef3a1a03669e0e821e7c83daf783',
              prettierPath: 'prettier',
              resetMocks: false,
              resetModules: false,
              resolver: undefined,
              restoreMocks: false,
              rootDir: 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend',
              roots: [ 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend' ],
              runner: 'jest-runner',
              setupFiles: [],
              setupFilesAfterEnv: [],
              skipFilter: false,
              skipNodeResolution: undefined,
              snapshotResolver: undefined,
              snapshotSerializers: [],
              testEnvironment: 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend\\node_modules\\jest-environment-jsdom\\build\\index.js',
              testEnvironmentOptions: {},
              testLocationInResults: false,
              testMatch: [ '**/__tests__/**/*.[jt]s?(x)', '**/?(*.)+(spec|test).[tj]s?(x)' ],
              testPathIgnorePatterns: [
                'C:\\\\Users\\\\Bruce\\\\Dropbox\\\\Code\\\\NODE BACKEND\\\\MoneyPlanBackend\\\\dist\\\\',
                'C:\\\\Users\\\\Bruce\\\\Dropbox\\\\Code\\\\NODE BACKEND\\\\MoneyPlanBackend\\\\node_modules\\\\'
              ],
              testRegex: [],
              testRunner: 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend\\node_modules\\jest-jasmine2\\build\\index.js',
              testURL: 'http://localhost',
              timers: 'real',
              transform: [ [ '.(js|jsx|ts|tsx)', 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend\\node_modules\\@sucrase\\jest-plugin\\dist\\index.js', {} ] ],
              transformIgnorePatterns: [ '\\\\node_modules\\\\' ],
              unmockedModulePathPatterns: undefined,
              watchPathIgnorePatterns: []
            }
          ],
          globalConfig: {
            bail: 0,
            changedFilesWithAncestor: false,
            changedSince: undefined,
            collectCoverage: false,
            collectCoverageFrom: [ 'src/**/*.js', '!src/**/*.debug.js', '!**/node_modules/**' ],
            collectCoverageOnlyFrom: undefined,
            coverageDirectory: 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend\\coverage',
            coverageProvider: 'babel',
            coverageReporters: [ 'json', 'text', 'lcov', 'clover' ],
            coverageThreshold: undefined,
            detectLeaks: undefined,
            detectOpenHandles: undefined,
            enabledTestsMap: undefined,
            errorOnDeprecated: false,
            expand: false,
            filter: undefined,
            findRelatedTests: false,
            forceExit: false,
            globalSetup: undefined,
            globalTeardown: undefined,
            json: false,
            lastCommit: false,
            listTests: undefined,
            logHeapUsage: false,
            maxConcurrency: 5,
            maxWorkers: 7,
            noSCM: undefined,
            noStackTrace: false,
            nonFlagArgs: undefined,
            notify: false,
            notifyMode: 'failure-change',
            onlyChanged: false,
            onlyFailures: false,
            outputFile: undefined,
            passWithNoTests: undefined,
            projects: [],
            replname: undefined,
            reporters: undefined,
            rootDir: 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend',
            runTestsByPath: false,
            silent: undefined,
            skipFilter: false,
            testFailureExitCode: 1,
            testNamePattern: undefined,
            testPathPattern: '',
            testResultsProcessor: undefined,
            testSequencer: 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend\\node_modules\\@jest\\test-sequencer\\build\\index.js',
            testTimeout: undefined,
            updateSnapshot: 'new',
            useStderr: false,
            verbose: undefined,
            watch: false,
            watchAll: false,
            watchPlugins: undefined,
            watchman: true
          },
          hasDeprecationWarnings: false,
          wallaby: {
            roots: [],
            watchPathIgnorePatterns: [ '\\\\node_modules\\\\', '\\.\\\\dist\\\\|\\.\\\\build\\\\|\\.\\\\coverage\\\\|\\\\\\..+\\\\', '\\\\dist\\\\' ],
            testPathIgnorePatterns: [ '\\\\node_modules\\\\', '\\.\\\\dist\\\\|\\.\\\\build\\\\|\\.\\\\coverage\\\\|\\\\\\..+\\\\', '\\\\dist\\\\' ],
            testMatch: [ '**/__tests__/**/*.[jt]s?(x)', '**/?(*.)+(spec|test).[tj]s?(x)' ],
            testRegex: []
          }
        }
      }
    },
    testFramework: { version: 'jest@24.8.0', configurator: 'jest@24.8.0', reporter: 'jest@24.8.0', starter: 'jest@24.8.0', autoDetected: true },
    filesWithCoverageCalculated: [ 'src/**/*.js', '!src/**/*.debug.js', '!**/node_modules/**' ],
    globalSetup: false,
    micromatch: true,
    files: [
      { pattern: '\\\\node_modules\\\\', regexp: /\\node_modules\\/, ignore: true, trigger: true, load: true },
      { pattern: '\\.\\\\dist\\\\|\\.\\\\build\\\\|\\.\\\\coverage\\\\|\\\\\\..+\\\\', regexp: /\.\\dist\\|\.\\build\\|\.\\coverage\\|\\\..+\\/, ignore: true, trigger: true, load: true },
      { pattern: '\\\\dist\\\\', regexp: /\\dist\\/, ignore: true, trigger: true, load: true },
      { pattern: '**/**', ignore: false, trigger: true, load: true, order: 1 },
      { pattern: '**/__tests__/**/*.[jt]s?(x)', ignore: true, trigger: true, load: true },
      { pattern: '**/?(*.)+(spec|test).[tj]s?(x)', ignore: true, trigger: true, load: true }
    ],
    tests: [
      { pattern: '\\\\node_modules\\\\', regexp: /\\node_modules\\/, ignore: true, trigger: true, load: true, test: true },
      {
        pattern: '\\.\\\\dist\\\\|\\.\\\\build\\\\|\\.\\\\coverage\\\\|\\\\\\..+\\\\',
        regexp: /\.\\dist\\|\.\\build\\|\.\\coverage\\|\\\..+\\/,
        ignore: true,
        trigger: true,
        load: true,
        test: true
      },
      { pattern: '\\\\dist\\\\', regexp: /\\dist\\/, ignore: true, trigger: true, load: true, test: true },
      { pattern: '**/__tests__/**/*.[jt]s?(x)', ignore: false, trigger: true, load: true, test: true, order: 2 },
      { pattern: '**/?(*.)+(spec|test).[tj]s?(x)', ignore: false, trigger: true, load: true, test: true, order: 3 }
    ],
    filesWithNoCoverageCalculated: [],
    runAllTestsInAffectedTestFile: false,
    updateNoMoreThanOneSnapshotPerTestFileRun: false,
    compilers: {},
    preprocessors: {},
    maxConsoleMessagesPerTest: 100,
    autoConsoleLog: true,
    delays: { run: 0, edit: 100, update: 0 },
    workers: { initial: 0, regular: 0, recycle: false },
    teardown: undefined,
    hints: {
      ignoreCoverage: '__REGEXP /ignore coverage|istanbul ignore/',
      ignoreCoverageForFile: '__REGEXP /ignore file coverage/',
      commentAutoLog: '?',
      testFileSelection: { include: '__REGEXP /file\\.only/', exclude: '__REGEXP /file\\.skip/' }
    },
    automaticTestFileSelection: true,
    runSelectedTestsOnly: false,
    extensions: {},
    env: { type: 'node', params: {}, runner: 'C:\\Program Files\\nodejs\\node.exe', viewportSize: { width: 800, height: 600 }, options: { width: 800, height: 600 }, bundle: true },
    reportUnhandledPromises: true,
    slowTestThreshold: 75,
    lowCoverageThreshold: 80,
    loose: true,
    configCode: 'auto.detect#575196754'
  },
  packageJSON: {
    dependencies: { 'base-64': '^0.1.0', 'big.js': '^5.2.2', dotenv: '^8.2.0', 'lodash.clonedeep': '^4.5.0', seedrandom: '^3.0.5', uuid: '^8.3.0' },
    devDependencies: {
      '@sucrase/jest-plugin': '^2.0.0',
      '@types/jest': '^25.1.4',
      'babel-eslint': '^10.1.0',
      eslint: '^6.8.0',
      'eslint-config-airbnb-base': '^14.0.0',
      'eslint-config-prettier': '^6.10.0',
      'eslint-plugin-import': '^2.20.1',
      'eslint-plugin-jest-formatting': '^1.2.0',
      'eslint-plugin-node': '^11.0.0',
      'eslint-plugin-promise': '^4.2.1',
      jest: '^25.1.0',
      nodemon: '^2.0.2',
      prettier: '^2.0.5',
      sucrase: '^3.12.1',
      xml2js: '^0.4.23'
    }
  },
  fs: { numberOfFiles: 104 },
  debug: [
    '2020-11-03T02:54:38.153Z angular/cli config Angular CLI not found.\n',
    '2020-11-03T02:54:38.271Z jest/config Detected Jest.\n',
    '2020-11-03T02:54:38.272Z jest/config Configured Jest.\n',
    '2020-11-03T02:54:38.273Z project Wallaby Node version: v12.18.3\n',
    '2020-11-03T02:54:38.273Z project Wallaby config: C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend\\auto.detect\n',
    '2020-11-03T02:54:39.309Z project File cache: C:\\Users\\Bruce\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.248\\projects\\d3e4866eedb79af9\n',
    '2020-11-03T02:54:39.348Z uiService Listening port 51235\n',
    '2020-11-03T02:54:39.383Z workers Parallelism for initial run: 6, for regular run: 3\n',
    '2020-11-03T02:54:39.383Z workers Starting run worker instance #0\n',
    '2020-11-03T02:54:39.384Z workers Starting run worker instance #1\n',
    '2020-11-03T02:54:39.384Z workers Starting run worker instance #2\n',
    '2020-11-03T02:54:39.384Z workers Starting run worker instance #3\n',
    '2020-11-03T02:54:39.384Z workers Starting run worker instance #4\n',
    '2020-11-03T02:54:39.384Z workers Starting run worker instance #5\n',
    '2020-11-03T02:54:39.385Z workers Web server is listening at 58256\n',
    '2020-11-03T02:54:39.421Z project File cache requires some updates, waiting required files from IDE\n',
    '2020-11-03T02:54:39.497Z project Stopping process pool\n',
    '2020-11-03T02:54:39.499Z project Test run started; run priority: 3\n',
    '2020-11-03T02:54:39.501Z project Running all tests\n',
    '2020-11-03T02:54:39.503Z workers Starting test run, priority: 3\n',
    '2020-11-03T02:54:39.503Z workers Distributing tests between 6 workers\n',
    '2020-11-03T02:54:39.503Z workers Running tests in parallel\n',
    '2020-11-03T02:54:39.504Z nodeRunner Starting sandbox [worker #0, session #55mu4]\n',
    '2020-11-03T02:54:39.504Z nodeRunner Starting sandbox [worker #1, session #gzb2d]\n',
    '2020-11-03T02:54:39.504Z nodeRunner Starting sandbox [worker #2, session #0sr2p]\n',
    '2020-11-03T02:54:39.504Z nodeRunner Starting sandbox [worker #3, session #8bqpl]\n',
    '2020-11-03T02:54:39.504Z nodeRunner Starting sandbox [worker #4, session #so7ks]\n',
    '2020-11-03T02:54:39.504Z nodeRunner Starting sandbox [worker #5, session #lqb1e]\n',
    '2020-11-03T02:54:39.504Z nodeRunner Preparing sandbox [worker #0, session #55mu4]\n',
    '2020-11-03T02:54:39.504Z nodeRunner Preparing sandbox [worker #1, session #gzb2d]\n',
    '2020-11-03T02:54:39.504Z nodeRunner Preparing sandbox [worker #2, session #0sr2p]\n',
    '2020-11-03T02:54:39.504Z nodeRunner Preparing sandbox [worker #3, session #8bqpl]\n',
    '2020-11-03T02:54:39.504Z nodeRunner Preparing sandbox [worker #4, session #so7ks]\n',
    '2020-11-03T02:54:39.504Z nodeRunner Preparing sandbox [worker #5, session #lqb1e]\n',
    '2020-11-03T02:54:39.544Z workers Started run worker instance (delayed) #0\n',
    '2020-11-03T02:54:39.544Z nodeRunner Prepared sandbox [worker #0, session #55mu4]\n',
    '2020-11-03T02:54:39.544Z workers [worker #0, session #55mu4] Running tests in sandbox\n',
    '2020-11-03T02:54:39.552Z workers Started run worker instance (delayed) #2\n',
    '2020-11-03T02:54:39.552Z nodeRunner Prepared sandbox [worker #2, session #0sr2p]\n',
    '2020-11-03T02:54:39.552Z workers [worker #2, session #0sr2p] Running tests in sandbox\n',
    '2020-11-03T02:54:39.553Z workers Started run worker instance (delayed) #1\n',
    '2020-11-03T02:54:39.553Z nodeRunner Prepared sandbox [worker #1, session #gzb2d]\n',
    '2020-11-03T02:54:39.554Z workers [worker #1, session #gzb2d] Running tests in sandbox\n',
    '2020-11-03T02:54:39.561Z workers Started run worker instance (delayed) #3\n',
    '2020-11-03T02:54:39.562Z nodeRunner Prepared sandbox [worker #3, session #8bqpl]\n',
    '2020-11-03T02:54:39.562Z workers [worker #3, session #8bqpl] Running tests in sandbox\n',
    '2020-11-03T02:54:39.564Z workers Started run worker instance (delayed) #4\n',
    '2020-11-03T02:54:39.564Z nodeRunner Prepared sandbox [worker #4, session #so7ks]\n',
    '2020-11-03T02:54:39.564Z workers [worker #4, session #so7ks] Running tests in sandbox\n',
    '2020-11-03T02:54:39.572Z workers Started run worker instance (delayed) #5\n',
    '2020-11-03T02:54:39.572Z nodeRunner Prepared sandbox [worker #5, session #lqb1e]\n',
    '2020-11-03T02:54:39.572Z workers [worker #5, session #lqb1e] Running tests in sandbox\n',
    '2020-11-03T02:54:42.806Z workers [so7ks] Loaded unknown number of test(s)\n',
    '2020-11-03T02:54:42.811Z workers [so7ks] Test executed: Handling Entries\n',
    '2020-11-03T02:54:42.822Z workers [so7ks] Run 1 test(s), skipped 0 test(s)\n',
    '2020-11-03T02:54:42.823Z workers [so7ks] Sandbox is responsive, closing it\n',
    '2020-11-03T02:54:42.906Z workers [0sr2p] Loaded unknown number of test(s)\n',
    '2020-11-03T02:54:42.908Z workers [0sr2p] Test executed: Handling Items\n',
    '2020-11-03T02:54:42.913Z workers [0sr2p] Loaded unknown number of test(s)\n',
    '2020-11-03T02:54:42.918Z workers [0sr2p] Test executed: should throw Error with correct name and message\n',
    '2020-11-03T02:54:42.928Z workers [0sr2p] Test executed: should throw Error with correct name and message\n',
    '2020-11-03T02:54:42.928Z workers [0sr2p] Test executed: should throw Error with correct name and message\n',
    '2020-11-03T02:54:42.943Z workers [0sr2p] Run 4 test(s), skipped 0 test(s)\n',
    '2020-11-03T02:54:42.944Z workers [0sr2p] Sandbox is responsive, closing it\n',
    '2020-11-03T02:54:43.502Z workers [lqb1e] Loaded unknown number of test(s)\n',
    '2020-11-03T02:54:43.502Z workers [lqb1e] Test executed: formats percent values into expected strings\n',
    '2020-11-03T02:54:43.502Z workers [lqb1e] Test executed: formats dollar values into expected string\n',
    '2020-11-03T02:54:43.502Z workers [lqb1e] Test executed: formats person birthdate\n',
    '2020-11-03T02:54:43.510Z workers [lqb1e] Test executed: clears All Persons TaxData \n',
    '2020-11-03T02:54:43.510Z workers [lqb1e] Test executed: counts Sources Of PayoutFunds \n',
    '2020-11-03T02:54:43.510Z workers [lqb1e] Test executed: tests some PlanEndYear considerations\n',
    '2020-11-03T02:54:43.511Z workers [lqb1e] Test executed: tests person life considerations\n',
    '2020-11-03T02:54:43.511Z workers [lqb1e] Test executed: tests plan fund balances \n',
    '2020-11-03T02:54:43.527Z workers [lqb1e] Run 8 test(s), skipped 0 test(s)\n',
    '2020-11-03T02:54:43.528Z workers [lqb1e] Sandbox is responsive, closing it\n',
    '2020-11-03T02:54:43.556Z workers [gzb2d] Loaded unknown number of test(s)\n',
    '2020-11-03T02:54:43.557Z workers [gzb2d] Test executed: Handling items\n',
    '2020-11-03T02:54:43.570Z workers [55mu4] Loaded unknown number of test(s)\n',
    '2020-11-03T02:54:43.571Z workers [gzb2d] Loaded unknown number of test(s)\n',
    '2020-11-03T02:54:43.571Z workers [55mu4] Test executed: Handling items\n',
    '2020-11-03T02:54:43.571Z workers [gzb2d] Test executed: tests the availablity of payout sources \n',
    '2020-11-03T02:54:43.580Z workers [55mu4] Loaded unknown number of test(s)\n',
    '2020-11-03T02:54:43.581Z workers [55mu4] Test executed: gets the named example planObject\n',
    '2020-11-03T02:54:43.583Z workers [55mu4] Test executed: moves specified parameter change\n',
    '2020-11-03T02:54:43.584Z workers [55mu4] Test executed: tests all assets to move any specified parameter change\n',
    '2020-11-03T02:54:43.585Z workers [55mu4] Test executed: adds a 2nd parameter change for Bobby College Payments\n',
    '2020-11-03T02:54:43.587Z workers [55mu4] Test executed: adds a 3rd parameter change for Bobby College Payments\n',
    '2020-11-03T02:54:43.589Z workers [55mu4] Test executed: adds a 2nd parameter change for global parameters\n',
    '2020-11-03T02:54:43.589Z fs File changed: src/__snapshots__/_modulePlanObject.test.js.snap\n',
    '2020-11-03T02:54:43.593Z workers [55mu4] Test executed: confirms existing parameter changes\n',
    '2020-11-03T02:54:43.593Z workers [55mu4] Test executed: consolidates ParamChangeDictionary of asset\n',
    '2020-11-03T02:54:43.601Z workers [gzb2d] Run 2 test(s), skipped 0 test(s)\n',
    '2020-11-03T02:54:43.602Z workers [gzb2d] Sandbox is responsive, closing it\n',
    '2020-11-03T02:54:43.606Z workers [8bqpl] Loaded unknown number of test(s)\n',
    '2020-11-03T02:54:43.606Z workers [8bqpl] Test executed: Handling Entries\n',
    '2020-11-03T02:54:43.607Z workers [55mu4] Run 9 test(s), skipped 0 test(s)\n',
    '2020-11-03T02:54:43.608Z workers [55mu4] Sandbox is responsive, closing it\n',
    '2020-11-03T02:54:43.611Z workers Sandbox (active) [8bqpl] error: planObject.parameters.theArray is not a function\n',
    '2020-11-03T02:54:43.613Z workers [8bqpl] Loaded unknown number of test(s)\n',
    '2020-11-03T02:54:43.615Z workers Sandbox (active) [8bqpl] error: Your test suite must contain at least one test.\n',
    '2020-11-03T02:54:43.633Z workers [8bqpl] Run 1 test(s), skipped 0 test(s)\n',
    '2020-11-03T02:54:43.634Z workers [8bqpl] Sandbox is responsive, closing it\n',
    '2020-11-03T02:54:43.635Z workers Failed to map the stack to user code, entry message: Your test suite must contain at least one test., stack: Error: Your test suite must contain at least one test.\n' +
      '    at onResult (C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend\\node_modules\\@jest\\core\\build\\TestScheduler.js:173:18)\n' +
      '    at C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend\\node_modules\\jest-runner\\build\\index.js:111:29\n',
    '2020-11-03T02:54:43.635Z workers Merging parallel test run results\n',
    '2020-11-03T02:54:43.645Z project Test run finished\n',
    '2020-11-03T02:54:43.645Z project Processed console.log entries\n',
    '2020-11-03T02:54:43.646Z project Processed loading sequences\n',
    '2020-11-03T02:54:43.646Z project Test name duplicate: should throw Error with correct name and message\n',
    '2020-11-03T02:54:43.646Z project Test name duplicate: should throw Error with correct name and message\n',
    '2020-11-03T02:54:43.646Z project Processed executed tests\n',
    '2020-11-03T02:54:43.672Z project Processed code coverage\n',
    '2020-11-03T02:54:43.948Z project Test run result processed and sent to IDE\n',
    '2020-11-03T02:54:43.960Z project Changes did not trigger any tests\n'
  ]
}
```

**Comment by ArtemGovorov** [comment]
Can you please run `npx jest` command in the repo's root to check if it's failing or passing?

If the `npx jest` run is passing, can you please:
- share show your `src/_modulePlanObject.test.js` looks like? You may obviously remove any sensitive data from the file, and just leave the imports and the failing test and the `describe/it` structure around it.

**Comment by martinduo** [comment]
I fixed it. Thanks to your suggestion, I found that npx jest failed also.
And that lead to further digging and I found that in just one of my first
written tests, _modulePlanObject.test.js, I had delayed the call to
addCrumbsPlusHelpers() to allow me to specifically test the result of that
call later. All other test files included that call to
addCrumbsPlusHelpers() in their initialization.

So that left the functions missing for any other call to
moduleObjectRoutines() that expected the functions to be present. That case
was not present until I added the usage of theArray() there later in the
course of writing more tests, and then Wallaby made that call as part of
its own initializations (I think).

I never thought to try npx jest, especially after I ran my VSCode debug
jest script and stepped through the code and found the function present.
That was successful probably due to a different sequence of preliminary
actions that added the functions. (My own "timing problem").

Again thanks for your help. As a retired developer working at home (NYC),
being able to walk to another programmer to ask about a coding problem is
what I miss most. That interaction always seemed to lead to finding a fix.
You supplied that help. Overnight too! Thanks.
